packages/jsx-parser/src/index.ts

```diff
@@ -30,10 +30,7 @@ export function createJSXParser(id: string = "solid-parser") {
   function childrenTokens<T>(children: () => JSXElement | JSXElement[] | T | T[]) {
     return createMemo(() =>
       children
-        ? ([] as any[])
-            .concat(children())
-            .filter(child => child && $TOKEN in child)
-            .map((a: T) => a as T)
```

Comment:
I was thinking that we could hide the 'attaching properties to a function'-implementation if we wanted:

``` ts
.map((a: T) => ({...a}))
```

then the result of `childrenTokens` would be an array of objects instead of an array of `createToken`.
but I am not sure if it's even a good thing to hide the implementation.

Reply:
It would also make new references each time. So bad idea.